### ISSUE TITLE
Potentially unwanted behaviour in ReplaceIgnoreCase

### ISSUE THREAD

**Issue by nicolaparo** [marcominerva/TinyHelpers#136]
The extension method  `public static string ReplaceIgnoreCase(this string input, string pattern, string replacement)` in src/TinyHelpers/Extensions/StringExtensions.cs is accepting a pattern parameter and is using is as a Regex to perform the case-insensitive replacement. However, the name of the method doesn't explicitly say that a regex will be used, and this might lead to potential unwanted results with some specific inputs. A possible implementation might be a facade to the method `public string Replace (string oldValue, string? newValue, StringComparison comparisonType);` where the comparisonType is `StringComparison.OrdinalIgnoreCase`, as it was done for the other methods in the same class, or sanitizing the pattern input with `Regex.Escape`. The `Replace` method overload that accepts the comparisonType in the string class is not available in netstandard2.0 but is part of netstandard2.1.

**Comment by nicolaparo** [comment]
Sure! here is an example done in net7.0:
```csharp
using TinyHelpers.Extensions;

Console.WriteLine("/path/to/file.*".ReplaceIgnoreCase(@".*", ".docx"));
Console.WriteLine("/path/to/file.*".Replace(@".*", ".docx", StringComparison.OrdinalIgnoreCase));
```
An user would expect the two printed lines to be the same, but the application actually prints
```
.docx.docx
/path/to/file.docx
```

By re-implementing the method as follows, it can resolve the problem in a netstandard2.0 compatible way:
```csharp
public static class StringExtensions
{
    public static string ReplaceIgnoreCase(this string input, string oldValue, string newValue) 
        => Regex.Replace(input, Regex.Escape(oldValue), newValue);
}
```

**Comment by marcominerva** [comment]
OK, I got the point. So, can you make a Pull Request? ;) And perhaps you can also add a test in the **Tests** project ;)

**Comment by marcominerva** [comment]
Published #138
